File: suscripciones.py
def crear_suscripcion(lista_suscripciones, nueva_suscripcion):
    return lista_suscripciones + [nueva_suscripcion]


def listar_suscripciones(lista_suscripciones):
    return lista_suscripciones


def suscripciones_activas(lista_suscripciones):
    return list(filter(lambda suscripcion: suscripcion["estado"] == "Activa", lista_suscripciones))


def cancelar_suscripcion(lista_suscripciones, usuario_id):
    return list(map(
        lambda suscripcion: 
        {**suscripcion, "estado": "Cancelada"} 
        if suscripcion["usuario_id"] == usuario_id else suscripcion,
        lista_suscripciones
    ))

File: suscripciones.go
package main

import "errors"

// Suscripcion representa el plan contratado por un usuario
type Suscripcion struct {
	id        int
	usuarioID int
	plan      string
	estado    string
}

// NuevaSuscripcion crea una suscripción y valida que el plan no esté vacío.
func NuevaSuscripcion(id int, usuarioID int, plan string) (Suscripcion, error) {
	if plan == "" {
		return Suscripcion{}, errors.New("el plan de suscripción es obligatorio")
	}

	return Suscripcion{
		id:        id,
		usuarioID: usuarioID,
		plan:      plan,
		estado:    "Activa",
	}, nil
}

// MostrarInfo devuelve el plan y estado actual de la suscripción.
func (s Suscripcion) MostrarInfo() string {
	return s.plan + " - " + s.estado
}
